docker-compose.yaml

```
@@ -2,23 +2,27 @@ services:
   server:
     build: 
       context: ./src/server
+      dockerfile: ./docker/Dockerfile
+      target: local
     container_name: federated-collection-discovery-server
     ports:
       - "8000:8000"
     volumes:
-      - ./src/server/app:/app/app
+      - ./src/server/app:/work/app
     environment:
       - PYTHONUNBUFFERED=1
       - FEDERATED_STAC_API_URLS=https://stac.maap-project.org/,https://openveda.cloud/api/stac/,https://catalogue.dataspace.copernicus.eu/stac
       # - FEDERATED_CMR_URLS=https://cmr.earthdata.nasa.gov/search/
-
   client:
-    build: ./src/client
+    build: 
+      context: ./src/client
+      dockerfile: ./docker/Dockerfile.dev
     container_name: federated-collection-discovery-client
     ports:
       - "3000:3000"
     volumes:
       - ./src/client:/app
+      - /app/node_modules
```

I recommend changing this to use a named volume so that when you run `docker volume ls` you get a nice name (e.g., `cross-catalog-search_node_modules`, if you follow my suggestion here, instead of some giant, meaningless name, such as `d92fcf9ecb6bd78dea66753f56f060449412f245be921aa092d040e9ed085791`, which doesn't give you any clue as to what it's for):

```suggestion
      - node_modules:/app/node_modules
```

Along with adding this top-level entry:

```
volumes:
  node_modules:
```

---

Actually, since I am running `yarn install` in the Dockerfile, does it even make sense to mount the `node_modules` directory? I guess it would make it possible to make changes to packages while the docker network is running without needing to rebuild the whole image.

---

Yes, it will speed things up during development after the initial build of the image. Since you copy the package.json into the container, if you change that file and rebuild the image, the rebuild will be much faster since `yarn install` will be much faster given an existing `node_modules` directory.